components/datepicker.tsx

```diff
@@ -0,0 +1,62 @@
+import { Field } from "formik";
+import DateView from "react-datepicker";
+
+export default function MonthYearDatePicker({ name, ...rest }: any) {
+  return (
+    <div>
+      <Field name={name}>
+        {({ form, field }: any) => {
+          const { setFieldValue } = form;
+          const { value } = field;
+
+          return (
+            <DateView
+              {...field}
+              {...rest}
+              selected={value}
+              onChange={(val: any) => setFieldValue(name, val)}
+              dateFormat="MMMM yyyy"
+              showMonthYearPicker
+              renderCustomHeader={({ date, changeYear }) =>
+                CustomYearDropdownPicker({ date, changeYear })
+              }
+            />
+          );
+        }}
+      </Field>
+    </div>
+  );
+}
+
+function CustomYearDropdownPicker({ date, changeYear }: any) {
+  const currentYear = new Date().getFullYear();
+  const years = range(currentYear - 20, currentYear).map((y) => ({
+    label: y.toString(),
+    value: y,
+  }));
+
+  return (
+    <select
+      value={date.getFullYear()}
+      onChange={(event) => {
+        changeYear(event.target.value);
+      }}
+    >
+      {years.map((y) => {
+        return (
+          <option key={y.value} value={y.value}>
+            {y.label}
+          </option>
+        );
+      })}
+    </select>
+  );
+}
+
+function range(start: number, end: number) {
+  const arr = [];
+  for (let i = start; i <= end; i++) {
+    arr.push(i);
+  }
+  return arr.reverse();
+}
```

Comment:
Well, instead of reversing it, you could just run the for backward 🤷🏻‍♂️

```suggestion
function range(start: number, end: number) {
  const arr = [];
  for (let i = end; i > start; i--) {
    arr.push(i);
  }
  return arr;
}
```

Reply:
I have 3 suggestions/options here:
- if logic of `range` is going to return reversed order, rename it to `rangeReverse` or something like that,
- leave it as `range` and return just `arr` without `reverse` and reverse it when you call `range(start, end)` so it's clear it's reversed, or
- create a function that will return array in descending order (e.g. 2023 to 2003) if `start` argument is greater than `end` argument e.g. `range(2023, 2023 - 20)` (this is definitely most complex one without any notable benefits as we need just simple array of years).